src/lib/Libutil/misc_utils.c

```diff
@@ -425,6 +425,63 @@ pbs_fgets_extend(char **pbuf, int *pbuf_size, FILE *fp)
 	return buf;
 }
 
+/**
+ * @brief
+ * 	Internal helper function for pbs_asprintf() to determine the length of post-formatted string
+ * 
+ * @param[in] fmt - printf format string
+ * @param[in] args - va_list arguments from pbs_asprintf()
+ * 
+ * @return int
+ * @retval length of post-formatted string
+ */
+int
+pbs_asprintf_len(const char *fmt, va_list args) 
+{
+	int len;
+#ifdef WIN32
+	len = _vscprintf(fmt, args);
+#else
+	{
+		va_list dupargs;
+		char c;
+
+		va_copy(dupargs, args);
+		len = vsnprintf(&c, 0, fmt, dupargs);
+		va_end(dupargs);
+	}
+#endif
+	return len;
+}
+
+/**
+ * @brief
+ * 	Internal helper function for pbs_asprintf() to allocate memory and format the string
+ * 
+ * @param[in] len - length of post-formatted string
+ * @param[in] fmt - format for printed string
+ * @param[in] args - va_list arguments from pbs_asprintf()
+ * 
+ * @return char *
+ * @retval formatted string in allocated buffer
+ */
+
+char *
+pbs_asprintf_format(int len, const char *fmt, va_list args)
```

Comment:
i'm not sure I like the name of this function. pbs_asprintf() also takes a format argument, so 'format' doesn't seem like the right distinction for this function. I actually think that pbs_asprintf_len() would be a good name for it since it needs the length as input. How about renaming the current "pbs_asprintf_len" to "get_str_va_len" and renaming "pbs_asprintf_format" to "pbs_asprintf_len" instead?

Reply:
I disagree.  The reason I named the functions the way I did was because of their purpose, not what they take.  pbs_asprintf_len() determines the length of the string.  It shouldn't be based on the fact that it takes a length parameter.  pbs_asprintf_format() actually does the formatting of the string.

Reply:
I personally got a bit confused because they are all "public" functions, so when one is choosing one of them, it didn't seem obvious to me which one I should be calling. But maybe it's just me, so I'll leave it up to you, naming is hard!